pkg/controllers/cspi-controller/handler.go

```diff
@@ -189,11 +199,11 @@ func (c *CStorPoolInstanceController) update(cspi *cstor.CStorPoolInstance) (*cs
 		WithKubeClientSet(c.kubeclientset).
 		WithOpenEBSClient(c.clientset).
 		WithRecorder(c.recorder)
-	cspi, err := oc.Update(cspi)
+	ncspi, err := oc.Update(cspi)
 	if err != nil {
-		return cspi, errors.Errorf("Failed to update pool due to %s", err.Error())
+		return ncspi, errors.Errorf("Failed to update pool due to %s", err.Error())
 	}
-	return c.updateStatus(cspi)
+	return c.updateStatus(ncspi)
 }
 
 func (c *CStorPoolInstanceController) updateStatus(cspi *cstor.CStorPoolInstance) (*cstor.CStorPoolInstance, error) {
```

Comment:
Can we rename this to `updateStatusCondition`

Reply:
I think it should be another way here we are really updating the entier status... func name in status.go will be updated to `updateStatusCondition`.

Reply:
Correct `UpdateStatus` needs to be renamed.

Reply:
Done